### PR TITLE
Bring back port mappings to the LB + Added Internal LB support

### PR DESCRIPTION
**Pull request by alena1108**

### CODE REVIEW

File: code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java [hunk 1]
```diff
@@ -6,66 +6,103 @@
...
+            if (!m.matches()) {
+                throw new ClientVisibleException(ResponseCodes.UNPROCESSABLE_ENTITY, WRONG_FORMAT);
+            }
+            if (m.group(1) != null) {
+                this.sourcePort = Integer.parseInt(m.group(1));
+            }
+            if (!StringUtils.isEmpty(m.group(2))) {
+                this.path = m.group(2);
+            }
+        } else {
```
Comment (ibuildthecloud) [inline code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java]: what if the split is length >= 4

Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java]: good point, will cover

File: code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java [hunk 2]
```diff
@@ -6,66 +6,103 @@
...
-            int slashIndex = domainPath.indexOf("/");
-            if (slashIndex == -1) {
-                this.domain = domainPath;
-            } else if (slashIndex == 0) {
-                this.path = domainPath;
-            } else {
-                this.domain = domainPath.substring(0, slashIndex);
-                this.path = domainPath.substring(slashIndex, domainPath.length());
+        if (splitted.length == 3) {
+            port = Integer.valueOf(splitted[2]);
```
Comment (ibuildthecloud) [inline code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java]: Is NumberFormatException caught anywhere?  Don't want to return a 500

Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java]: will add the catch

**Comment by alena1108** [comment]
Added internal lb support to the same PR

File: code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java [hunk 3]
```diff
@@ -3,69 +3,134 @@
...
+ * 80/path=81
+ * 80=81
+ * /path
+ * /path=81
+ * 81
+ */
 public class LoadBalancerTargetPortSpec {
-    public static final String WRONG_FORMAT = "PortWrongFormat";
-    public static final String INVALID_PORT = "PortInvalidPublicPort";
+    public static final String WRONG_FORMAT = "InvalidPort";
```
Comment (ibuildthecloud) [inline code/iaas/model/src/main/java/io/cattle/platform/core/util/LoadBalancerTargetPortSpec.java]: Why is WRONG_FORMAT = "InvalidPort"?

**Pull request merged**